### ISSUE TITLE
Multiarch (arm64 specifically) support, please

### ISSUE THREAD

**Issue by dbaker-rh** [halverneus/static-file-server#56]
This is the perfect web server for a tiny raspberry pi based kubernetes cluster.  It's a shame the images are amd64 only, so won't run on raspberry pi.

I did some hacking on it, and made the tiniest fork to create a multi-arch image for my own use.

The diff is two lines in the Dockerfile and some extra stuff needed to build and push the container images.

Notes on how to do it are here: https://github.com/dbaker-rh/static-file-server/blob/master/multiarch.md

**Comment by halverneus** [comment]
I'm reading through your proposal right now. I'm assuming you are referring to making a multi-arch builds that can be deployed from Docker Hub? That is something I am interested in adding (I originally applied for static-file-server to be an official Docker image, which would have supported five Linux architectures, but it was denied), but I haven't seen (nor looked at in the last two years) any way to do that by building on Docker Hub.

Outside of Docker, I do build for seven architectures when I issue a release (I can add more, if there is interest). Those builds use the Dockerfile.all file in the root of the repository.

I'll read what you sent and see about integrating it. Thanks!

**Comment by dbaker-rh** [comment]
Yes, this is for a multi-arch container image.  Docker hub is just an example of where to host it - quay.io does the same thing, but sometimes behaves a little better.  I've not tested either one to do my builds for me, and "podman push" what I want.

I uploaded my multi-arch image to the two public registries to kick the tires on this, mostly because I couldn't run the pod on my rPi and needed a quick fix.  Docker hub still isn't showing details -- https://hub.docker.com/r/dbakerrh/static-file-server/tags and quay.io decided it couldn't scan the amd64 image (which is unusual, since it certainly can scan those images) -- https://quay.io/repository/dbaker/static-file-server?tab=tags


Perhaps I breezed past the Dockerfile.all too quickly to begin with.  I see that it's building the binaries, but they aren't terribly useful being left in the builder container.  The positive is you confirm cross compilation can succeed for each.

If you're interested in updating your own copy to be multi-arch the notes in my .md file will definitely get you started.


In looking through the details I noticed a typo on your https://hub.docker.com/r/halverneus/static-file-server page.  Search the text for "Leave black" (should be "Leave blank").

**Comment by halverneus** [comment]
Yeah, the Dockerfile.all is used to create the binaries that are included with each Release on GitHub and isn't intended for any other use, at the moment. I do think I'm going to include the Docker arguments to set GOARCH and (if we are doing that, then it only makes sense to include) GOOS. That'll be enough for people to build it for their desired platforms. I'll do some research, though, to see if I can set up automated building to include other architectures on Docker Hub so that building will not be necessary.

**Comment by bostrt** [comment]
I'm looking forward for multiarch support for a home raspi server as well

**Comment by halverneus** [comment]
![image](https://user-images.githubusercontent.com/10501044/105620055-befe6300-5dad-11eb-8a94-a039af13909d.png)

I just added a few architecture variants directly to [Docker Hub](https://hub.docker.com/r/halverneus/static-file-server) as version 1.8.3 (you will need to pull halverneus/static-file-server:v1.8.3, as this doesn't apply to earlier versions, latest nor dev). Please report back if this meets your requirements and I'll formalize the build process. If it doesn't meet your needs, let me know what is missing and I'll adjust accordingly.

**Comment by bostrt** [comment]
Thanks! I'm pull the armv7 one and it runs :)

**Comment by halverneus** [comment]
That's great! I'll formalize the build process and all future releases will include builds for those three architectures.
